### DISCUSSION TITLE
[FE] react-query hook분리와 그 네이밍에 관하여

### DISCUSSION THREAD

**Discussion by kangyeongmin** [woowacourse-teams/2023-stamp-crush#658] [General]
페이지 단의 react-query 통신로직이 점점 많아지고 복잡해짐에 따라 /pages/[pageName/hooks 폴더 안에 관련 useQuery 혹은 useMutation을 훅으로 분리하기로 하였는데요!

현재 그 네이밍에 대한 방식이 모두 다 달라서 관련 컨벤션 도입이 필요해보여요.

**Comment by 2yunseong** [comment]
저는 직접 네이밍을 지었습니다~~

**Comment by regularPark** [comment]
훅에 사용하는 `use-` prefix에 기존의 통신코드를 정의할 때 사용했던 `메소드 + 기능`을 합쳐 

ex) `useGetCoupon` 

이런 방식으로 정의하면 어떨까요~?

**Comment by kangyeongmin** [comment]
결정된 사안: `use+[HTTPmethod]+[메서드가하는일]`

react-Query로 분리한 훅에서 해당 api가 어떤 요청인지(GET, PUT, DELETE, PATCH)인지도 알 필요가 있다고 생각했습니다. 
따라서, 레고의 방식으로 결정!
